db/compaction/compaction_service_job.cc

```diff
@@ -331,19 +348,33 @@ Status CompactionServiceCompactionJob::Run() {
     status = io_s;
   }
 
-  // Finish up all book-keeping to unify the subcompaction results
-  compact_->AggregateCompactionStats(compaction_stats_, *compaction_job_stats_);
-  UpdateCompactionStats();
-  RecordCompactionIOStats();
-
   LogFlush(db_options_.info_log);
   compact_->status = status;
   compact_->status.PermitUncheckedError();
 
-  // Build compaction result
+  // Build Compaction Job Stats
+
+  // 1. Aggregate CompactionOutputStats into Internal Compaction Stats
```

Comment:
Nit: Comment is a bit confusing as CompactionOutputStats is not a defined type.

Reply:
Inside `compact_->AggregateCompactionStats()` it calls `sc.AggregateCompactionOutputStats(compaction_stats);` (which I renamed from `AggregateCompactionStats()`), and `AggregateCompactionOutputStats()` calls `compaction_stats.stats.Add(compaction_outputs_.stats_);` and `compaction_outputs_.stats_` is actually `  InternalStats::CompactionOutputsStats` type. 

https://github.com/facebook/rocksdb/blob/8ad4c7efc48d301f5e85467105d7019a49984dc8/db/compaction/compaction_outputs.h#L310

As you've mentioned, it's honestly very confusing. I do agree that we should invest our time to re-organize and improve this. There was one comment Siying left two years ago and I hope to address it at some point. (https://github.com/facebook/rocksdb/pull/9964#pullrequestreview-986873927)